url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/WelcomeFrame.java

```diff
@@ -0,0 +1,141 @@
+
+	private JFrame frame;
+	private DataStore dataStore;
+	/**
+	 * Create the application.
+	 */
+	public WelcomeFrame() {
+		super();
+		System.out.println("new WelcomeFrame()");
+		setTitle("EdT");
+		initialize();
+		this.setVisible(true);
+	}
+
+	/**
+	 * Initialize the contents of the frame.
+	 */
+	private void initialize() {
+		this.setBounds(100, 100, 250, 140);
+		this.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		
+		
+		JButton b1 = new JButton("Nouveau");
+		b1.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				// On demande d'abord où l'utilisateur souhaite enregistrer sa base de données.
+				JFileChooser fc = new JFileChooser();
+				fc.setFileFilter(new FileNameExtensionFilter("Emploi du Temps", "EdT"));
+				int res = fc.showSaveDialog(frame);
+				File f = null;
+				if (res == JFileChooser.APPROVE_OPTION)	{
+					// On a le fichier.
+					f = fc.getSelectedFile();
+					// On sauvegarde sa localisation pour la prochaine ouverture du programme.
+					File dataStoreLocation = new File(System.getProperty("user.dir") + "/dataStoreLocation.loc");
+					BufferedWriter writer;
+					try {
+						writer = new BufferedWriter(new FileWriter(dataStoreLocation));
+						writer.write (f.getAbsolutePath()); // On écrit le chemin du DataStore
+						writer.close();
+						new StartFrame();
+					} catch (IOException e1) {
+						// TODO Auto-generated catch block
+						JOptionPane.showMessageDialog(null, "Impossible de sauvegarder la localisation du fichier de sauvegarde.", "Erreur de sauvegarde", JOptionPane.ERROR_MESSAGE);
+					}
+					setVisible(false);
+				}
+			}
+		});
+		
+		
+		JButton b2 = new JButton("Ouvrir");
+		b2.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				JFileChooser fc = new JFileChooser();
+				fc.setFileFilter(new FileNameExtensionFilter("Emploi du Temps", "EdT"));
+				int res = fc.showOpenDialog(frame);
+				File f = null;
+				if (res == JFileChooser.APPROVE_OPTION)	{
+					// On a le fichier.
+					f = fc.getSelectedFile();
+					// On sauvegarde sa localisation pour la prochaine ouverture du programme.
+					File dataStoreLocation = new File(System.getProperty("user.dir") + "/dataStoreLocation.loc");
+					BufferedWriter writer;
+					try {
+						writer = new BufferedWriter(new FileWriter(dataStoreLocation));
+						writer.write (f.getAbsolutePath()); // On écrit le chemin du DataStore
+						writer.close();
+					} catch (IOException e1) {
+						// TODO Auto-generated catch block
+						JOptionPane.showMessageDialog(null, "Impossible de sauvegarder la localisation du fichier de sauvegarde.", "Erreur de sauvegarde", JOptionPane.ERROR_MESSAGE);
+					}
+
+					// On essaie maintenant de charger le DataStore
+					try
+					{
+						FileInputStream inputFileStream = new FileInputStream(f.getAbsolutePath());
+						ObjectInputStream objectInputStream = new ObjectInputStream(inputFileStream);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *objectInputStream*. The resource is referenced at lines 100, 101. The resource is closed at lines 102, 103. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception. Ensure *objectInputStream* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).